bob/src/api/http.rs

```diff
@@ -485,6 +494,34 @@ async fn read_directory_children(mut read_dir: ReadDir, name: String, path: Stri
     }
 }
 
+#[get("/data/<key>")]
+fn get_data(bob: State<BobServer>, key: BobKey) -> Result<Content<Vec<u8>>, StatusExt> {
+    let opts = BobOptions::new_get(None);
+    let result = runtime()
+        .block_on(async { bob.grinder().get(key, &opts).await })
+        .map_err(|err| -> StatusExt { err.into() })?;
+    Ok(Content(infer_data_type(&result), result.inner().to_owned()))
+}
+
+#[post("/data/<key>", data = "<data>")]
+fn put_data(bob: State<BobServer>, key: BobKey, data: Data) -> Result<StatusExt, StatusExt> {
+    let mut data_buf = vec![];
+    data.open()
+        .read_to_end(&mut data_buf)
+        .map_err(|err| -> StatusExt { err.into() })?;
+    let data = BobData::new(
+        data_buf,
+        BobMeta::new(chrono::Local::now().timestamp() as u64),
+    );
+
+    let opts = BobOptions::new_put(None);
+    runtime()
```

Comment:
we now have runtime handler in `BobServer` so you should run async tasks with `bob.block_on(async { . . })`
Here and in other places below